### PR TITLE
add support show replica_query rule

### LINKED ISSUE #8936: Support "show replica_query rule"
## Feature Request
add support `show replica_query rule`

### Describe the feature you would like.
add support `show replica_query rule`, the result like this:

```
+-----------------------------+----------------------------------+----------------------------------+
|logicDataSourcename | primaryDataSourceName| replicaDataSourceNames|
+-----------------------------+----------------------------------+----------------------------------+
| xxxx                            | xxxxx                                | xxx                                    |
+-----------------------------+----------------------------------+----------------------------------+
```

### PR DESCRIPTION
**Pull request by jingshanglu**
Fixes #8936 .

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import java.util.Optional;
+
+/**
+ * Backend handler for show replica query rule.
+ */
+public final class ReplicaQueryRuleQueryBackendHandler extends SchemaRequiredBackendHandler<ShowRuleStatement> {
+    
+    private Iterator<ReplicaQueryDataSourceRuleConfiguration> data;
+    
+    private String schema;
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java:47]: Can it use final here?

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        }
+    }
+    
+    @Override
+    protected ResponseHeader execute(final String schemaName, final ShowRuleStatement sqlStatement) {
+        loadRuleConfiguration(schema);
+        return new QueryResponseHeader(getQueryHeader());
+    }
+    
+    private List<QueryHeader> getQueryHeader() {
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java:64]: Private method should be just next to the method in which it is used

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java (hunk 1 above)
Comment (jingshanglu) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java:47]: ok

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java (hunk 2 above)
Comment (jingshanglu) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java:64]: ok

Comment (terrymanu) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java:64]: loadRuleConfiguration should before getQueryHeader

Comment (jingshanglu) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/impl/ReplicaQueryRuleQueryBackendHandler.java:64]: ok

**Review by terrymanu — APPROVED** [review]

**Pull request merged**